pkg/azure/actuator.go

```diff
@@ -504,7 +504,7 @@ func (a *Actuator) syncMint(ctx context.Context, cr *minterv1.CredentialsRequest
 		return err
 	}
 
-	if clientSecret == "" {
+	if newClientSecret != "" {
```

Comment:
Took me a while to follow the logic here, but this looks like it would work. :+1: 

Should we error after this if block if clientSecret happens to still be "" ? (just out an abundance of caution even though the current code flow should never allow for it)

Reply:
yes, sounds good!